### ISSUE TITLE
Transform `SatelliteImage` class into an optional dictionary attribute in `Raster`

### ISSUE THREAD

**Issue by rhugonnet** [GlacioHack/geoutils#532]
After more thinking, the organization `Raster` / `SatelliteImage` / `DEM` does not make so much sense anymore: some DEMs can be purely modelled, and are not necessarily from an acquisition.

In the end, the `SatelliteImage` class is essentially just an extra metadata parser on top of `Raster`, which could live as simply an `__init__` option of the `Raster` class and store the metadata in an optional dictionary attribute such as the `tags` attribute for raster metadata.

In short, we could:
- Remove the `SatelliteImage` class,
- Add the functionalities as a `parse_sensor_metadata` option to `Raster`, true by default, which stores all in a `.sensor` dictionary attribute,
- Subclass `DEM` from `Raster`.

**Comment by adehecq** [comment]
Good idea !!
